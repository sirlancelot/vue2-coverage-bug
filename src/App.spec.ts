import { shallowMount } from "@vue/test-utils"
import Component from "./App.vue"

const createWrapper = () => shallowMount<any>(Component, {
	stubs: ["HelloWorld", "v-app", "v-main", "v-container"],
})

it("mounts", () => {
	const wrapper = createWrapper()
	expect(wrapper.html()).toMatchSnapshot()
})

it("tests coveredFn", () => {
	const wrapper = createWrapper()
	const consoleLog = vi.spyOn(console, "log").mockReturnValue()

	wrapper.vm.coveredFn()

	expect(consoleLog).toHaveBeenNthCalledWith(1, "this function is covered")
})
