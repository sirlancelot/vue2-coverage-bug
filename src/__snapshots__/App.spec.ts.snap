// Vitest Snapshot v1

exports[`mounts 1`] = `
"<v-app-stub id=\\"app\\">
  <v-main-stub>
    <v-container-stub><img alt=\\"Vue logo\\" src=\\"/src/assets/logo.png?import\\">
      <helloworld-stub msg=\\"Hello Vue 2 + TypeScript + Vite\\"></helloworld-stub>
    </v-container-stub>
  </v-main-stub>
</v-app-stub>"
`;
